### ISSUE TITLE
dotnet.wasm in node.js looks for module called 'a' and fails

### ISSUE THREAD

**Issue by am11** [dotnet/runtime#41861]
`dotnet.wasm` module does not load cleanly in the node.js environment. It complains about a missing `package 'a'`.

Repro:
```sh
$ node -v
v14.9.0

$ cat > test.mjs << EOF
import * as dotnet from '/Users/am11/.nuget/packages//microsoft.netcore.app.runtime.browser-wasm/5.0.0-rc.1.20425.18/runtimes/browser-wasm/native/dotnet.wasm';

const listMethods = subject => Object.getOwnPropertyNames(subject).filter(item => typeof subject[item] === 'function');

console.log(listMethods(dotnet));
EOF

$ node --experimental-modules --experimental-wasm-modules test.mjs
internal/process/esm_loader.js:74
    internalBinding('errors').triggerUncaughtException(
                              ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find package 'a' imported from /Users/am11/.nuget/packages/microsoft.netcore.app.runtime.browser-wasm/5.0.0-rc.1.20425.18/runtimes/browser-wasm/native/dotnet.wasm
    at packageResolve (internal/modules/esm/resolve.js:623:9)
    at moduleResolve (internal/modules/esm/resolve.js:664:18)
    at Loader.defaultResolve [as _resolve] (internal/modules/esm/resolve.js:778:11)
    at Loader.resolve (internal/modules/esm/loader.js:85:40)
    at Loader.getModuleJob (internal/modules/esm/loader.js:229:28)
    at ModuleWrap.<anonymous> (internal/modules/esm/module_job.js:51:40)
    at link (internal/modules/esm/module_job.js:50:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}
```

Similar error about `module 'a'` occurs if we try [`WebAssembly.instantiate`](https://developer.mozilla.org/en-US/docs/WebAssembly/Using_the_JavaScript_API) API:

```sh
$ cat > test2.mjs << EOF
import * as fs from 'fs';

const name = '/Users/am11/.nuget/packages//microsoft.netcore.app.runtime.browser-wasm/5.0.0-rc.1.20425.18/runtimes/browser-wasm/native/dotnet.wasm'
const bytes = fs.readFileSync(name);
const result =  WebAssembly.instantiate(bytes, {
  imports: { imported_func: arg => console.log(arg) }
}).catch(err => console.error(err));
EOF

$ node --experimental-modules --experimental-wasm-modules test2.mjs
[TypeError: WebAssembly.instantiate(): Import #0 module="a" error: module is not an object or function]
```

**Comment by am11** [comment]
Made some headway with Deno environment, instead of node.js, which offers a mature and non-experimental WASM support. Needed a build option to compile dotnet.js as an importable ES6 module: #41964.

**Comment by pavelsavara** [comment]
cc @Daniel-Genkin

**Comment by Daniel-Genkin** [comment]
@am11 I have been working on NodeJS support and got it working with the console sample. 
Here it is for reference: https://github.com/dotnet/runtime/pull/54640
Hope this is helpful

**Comment by am11** [comment]
Note that deno has since acquired utf-16le support, so `sed` hack in the PR description is not required anymore.

Opinion time:
1. I think we should consider naming the TargeOS as `Server` (as opposed to existing `Browser`) for node.js-like environments.
2. we should consider using JS modules unconditionally, since all environments that support wasm also support modules. (one less switch to test / worry about).

**Comment by SamMonoRT** [comment]
Assigning to @Daniel-Genkin for tracking

**Comment by SamMonoRT** [comment]
@lewing  @thaystg @Daniel-Genkin - is the PR going to be merged prior to 6.0 cutoff ? Please update the milestone accordingly. Thank you
